geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/misc/WANSSLDUnitTest.java

```diff
@@ -14,107 +14,624 @@
+  private int sendingLocatorPort, receivingLocatorPort;
+  private String senderId = "ln";
+
+  @Parameterized.Parameters(name = "dispatcherThreadCount={0}")
+  public static Collection<Integer> data() {
+    return new ArrayList<>(Arrays.asList(1, 3, 5));
+  }
+
+  @Parameterized.Parameter
+  public int dispatcherThreadCount;
+
+  /** This VM's connection to the distributed system */
+  private static InternalDistributedSystem system;
+  private String testMethodName;
+  private String replicatedRegionName;
+
+  @Rule
+  public SerializableTestName testMethodNameWatcher = new SerializableTestName();
+
+  @Rule
+  public ClusterStartupRule csRule = new ClusterStartupRule();
+
+  @Before
+  public void before() {
+    sendingLocator = csRule.getVM(0);
+    receivingLocator = csRule.getVM(1);
+    receivingServer = csRule.getVM(2);
+    sendingServer = csRule.getVM(3);
+
+    sendingLocatorPort = sendingLocator.invoke(() -> createFirstLocatorWithDSIdOne());
+    receivingLocatorPort = receivingLocator.invoke(() -> createFirstRemoteLocatorWithDSIdTwo(sendingLocatorPort));
+
+    // setup distributed test case
+    // Make sure all VMs agree on nomenclature
+    final String className = getClass().getCanonicalName();
+    testMethodName = getSanitizedCurrentTestName();
+    replicatedRegionName = testMethodName + "_RR";
+    String methodName = testMethodName;
+    setUpVM(getDefaultDiskStoreName(0, -1, className, testMethodName));
+
+    for (int hostIndex = 0; hostIndex < Host.getHostCount(); hostIndex++) {
+      Host host = Host.getHost(hostIndex);
+      for (int vmIndex = 0; vmIndex < host.getVMCount(); vmIndex++) {
+        final String vmDefaultDiskStoreName =
+            getDefaultDiskStoreName(hostIndex, vmIndex, className, testMethodName);
+        host.getVM(vmIndex).invoke("setupVM", () -> {
+          testMethodName = methodName;
+          replicatedRegionName = methodName + "_RR";
+          setUpVM(vmDefaultDiskStoreName);
+        });
+      }
+    }
 
-  public WANSSLDUnitTest() {
-    super();
+    // setup WAN test case
+    int nThreads = dispatcherThreadCount;
+    Invoke.invokeInEveryVM(() -> dispatcherThreadCount = nThreads);
+
+    IgnoredException.addIgnoredException("Connection refused");
+    IgnoredException.addIgnoredException("Software caused connection abort");
+    IgnoredException.addIgnoredException("Connection reset");
   }
 
-  @Test
-  public void testSenderSSLReceiverSSL() {
-    Integer lnPort = (Integer) vm0.invoke(() -> WANTestBase.createFirstLocatorWithDSId(1));
-    Integer nyPort = (Integer) vm1.invoke(() -> WANTestBase.createFirstRemoteLocator(2, lnPort));
+  private Integer createFirstLocatorWithDSIdOne() {
+    int port = AvailablePortHelper.getRandomAvailablePortForDUnitSite();
+    startLocator(1, port, port, -1);
+    return port;
+  }
 
-    vm2.invoke(() -> WANTestBase.createReceiverWithSSL(nyPort));
+  private Integer createFirstRemoteLocatorWithDSIdTwo(int remoteLocPort) {
```

Comment:
It isn't strictly necessary, but you could refactor this and createFirstLocatorWithDSIdOne to be one method with parameters of DSID and remoteLocPort, which would just take -1 for the first one.